Deprecate Domain `expires_on` in favor of `expires_at`

duduribeiro opened the pull request:
We deprecated expires_on attribute in our documentation and api in
favor of expires_at. This commit introduces the new field and removes
the old one. It also updates to use the new fixtures.

We should do a major bump, since we are removing an attribute.

Also synced these new fixtures with the docs one

duduribeiro commented on lib/dnsimple/domain_registration.ex:
```diff
@@ -20,6 +20,6 @@ defmodule Dnsimple.DomainRegistration do
 
   defstruct ~w(id domain_id registrant_id
                period state auto_renew whois_privacy
-               expires_on created_at updated_at)a
```
`expires_on` does not exist in DomainRegistration.

jacegu commented on CHANGELOG.md:
```diff
@@ -1,5 +1,9 @@
 
+## master
+
+- CHANGED: `Dnsimple.Domain` struct now have `expires_at` (timestamp) instead of `expires_on` (date only).
```
```suggestion
- CHANGED: `Dnsimple.Domain` struct now has `expires_at` (timestamp) instead of `expires_on` (date only).
```

jacegu commented:
@duduribeiro the only thing to keep in mind before releasing this is that clients should remain as much in sync as possible. So, [if we end up not removing `expires_on` in the Ruby client](https://github.com/dnsimple/dnsimple-ruby/pull/186), we should do the same in this one.

duduribeiro commented:
@jacegu and @OleMchls since I made change, wanted a new 👁️ on it.

@OleMchls with this new change, we will follow the same as we are doing in the other libraries, to not make a hard move and don't have a major bump.

weppos commented:
@duduribeiro the behavior in all the clients must be consistent. So this PR (and all the other clients) needs to be updated to behave as https://github.com/dnsimple/dnsimple-ruby/pull/186

duduribeiro commented:
@weppos @jacegu @OleMchls 
Recently we dropped the domain expires_on on dnsimple-go by suggestion of @weppos, because dnsimple-go is not yet in 1.x, it wouldn’t be a problem to make another major release.
now on dnsimple-elixir, it is already at 2.0.0, we made a recent major release, so I made a little bit different from the dnsimple-go https://github.com/dnsimple/dnsimple-elixir/pull/135
I kept the expires_on, added a comment telling it is deprecated and added the expires_at. do you agree with it?

jacegu commented on lib/dnsimple/domain.ex:
```diff
@@ -7,6 +7,7 @@ defmodule Dnsimple.Domain do
   - https://developer.dnsimple.com/v2/domains/#domain-attributes
   """
 
+  @doc "expires_on is deprecated in favor of expires_at"
```
@duduribeiro please note that Elixir has a `@deprecated` attribute for this https://hexdocs.pm/elixir/writing-documentation.html#documentation-metadata. I have never tried it on a Struct field though. You may need to check how it behaves there.

duduribeiro replied:
I didn't find a way to deprecate only the expires_on. The way it worked was possible only to deprecate the whole type.
